homeassistant/components/jewish_calendar/binary_sensor.py

```diff
@@ -0,0 +1,66 @@
+"""Support for Jewish Calendar binary sensors."""
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorDevice
+import homeassistant.util.dt as dt_util
+
+from . import DOMAIN, SENSOR_TYPES
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the Jewish Calendar binary sensor devices."""
+    if discovery_info is None:
+        return
+
+    async_add_entities(
+        [
+            JewishCalendarBinarySensor(hass.data[DOMAIN], sensor, sensor_info)
+            for sensor, sensor_info in SENSOR_TYPES["binary"].items()
+        ]
+    )
+
+
+class JewishCalendarBinarySensor(BinarySensorDevice):
+    """Representation of an Jewish Calendar binary sensor."""
+
+    def __init__(self, data, sensor, sensor_info):
+        """Initialize the binary sensor."""
+        self._location = data["location"]
+        self._type = sensor
+        self._name = f"{data['name']} {sensor_info[0]}"
+        self._icon = sensor_info[1]
+        self._hebrew = data["language"] == "hebrew"
+        self._candle_lighting_offset = data["candle_lighting_offset"]
+        self._havdalah_offset = data["havdalah_offset"]
+        self._state = False
+
+    @property
+    def icon(self):
+        """Return the icon of the entity."""
+        return self._icon
+
+    @property
+    def name(self):
+        """Return the name of the entity."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return true if sensor is on."""
+        return self._state
+
+    async def async_update(self):
+        """Update the state of the sensor."""
+        import hdate
```

Comment:
Please move 3rd party imports to the top of the module.

Reply:
Done